docs/proposals/20200330-spot-instances.md

```diff
@@ -262,6 +263,73 @@ This would be preferable as a DaemonSet would not be required on workload cluste
 Since this is not essential for running on non-guaranteed instances and existing solutions exist for each provider,
 users can deploy these existing solutions until CAPI has capacity to implement a solution.
 
+#### Termination handler design
+
+To enable graceful termination of workloads running on non-guaranteed instances,
+a termination handler pod will need to be deployed on each spot instance.
+
+The termination pod will be developed to poll the metadata service for the Node
+that it is running on.
+We will implement request/response handlers for each of the three cloud providers
+that will enable the handler to determine if the cloud provider instance is due
+for termination.
+
+The code for all providers will be mostly similar, the only difference is logic for polling termination endpoint and processing the response.
+This means that cloud provider type can be passed as an argument and termination handler can be common for all providers and placed in a separate repository, for example: `kubernetes-sigs/cluster-api-termination-handler`.
```

Comment:
Given that this doesn't actually exist today, should we remove it?

Reply:
Do you have a POC of what the code would look like?

Reply:
Yes, I'll put it all together and demonstrate it, probably next week.

Reply:
@CecileRobertMichon Hi, I made a prototype that should work https://github.com/alexander-demichev/termination-handler.